niri-ipc/src/lib.rs

```diff
@@ -523,6 +523,8 @@ pub struct Workspace {
     pub output: Option<String>,
     /// Whether the workspace is currently active on its output.
     pub is_active: bool,
+    /// The windows currently on this workspace
+    pub windows: Vec<Window>,
```

Comment:
This is one of the things that I'm not sure what the best design is. We have a `MappedId` now which we can add to `niri_ipc::Window` (maybe not right away, need to think some more if it should be u32 or u64). So we can have this as essentially `pub windows: Vec<MappedId>`, thereby avoiding sending a lot of extra data (`niri_ipc::Window` will surely grow in size in the future) along with every workspace. Then there can be another IPC call to get window info. This will be similar to how `output` is `Option<String>` and not `Option<Output>`.

Reply:
It does mean there's a race between niri msg workspaces and niri msg windows (or something), but then there will be the event stream IPC to solve that cleanly

Reply:
Oh yeah, that sounds like a pretty good solution, I'm not sure what the overhead of IPC is but having too large things sent over doesn't seem ideal